src/lib/flpTokenService.ts

```diff
@@ -0,0 +1,158 @@
+import { aoInstance } from "@/lib/aoconnect";
+import { redis } from "./redis";
+import { retryWithDelay } from "@/utils/retry.utils";
+
+interface RawFlpToken {
+  flp_token_name: string;
+  flp_token_ticker: string;
+  flp_token_denomination: number;
+  flp_token_logo: string;
+  flp_token_process: string;
+  flp_id: string;
+}
+
+interface FlpToken {
+  name: string;
+  ticker: string;
+  denomination: number;
+  logo: string;
+  id: string;
+  flpId: string;
+  autoClaim: boolean;
+}
+
+type DelegationRecord = Record<string, number>;
+
+interface CachedFlpTokenData {
+  flpTokens: FlpToken[];
+  timestamp: number;
+}
+
+const testTokenFlpIds = new Set([
+  "T3M4QSF7VGa0le7KtxBDHOaIcjnZeC-SQ7nh3ABuufs",
+  "So2HpldZaaVFbeH8mUGGzQBVdEzAx5HvMyMaZ47az_M",
+  "4mowY7A-b6WJyVR-Tde2m3Zcl_JVxil21c15PXiHhfA",
+  "-ntvNGm4onpKXS8SZ6-5sFnmjRHfMNAwS_JuR-pO504",
+  "WRkDu1hOeNksAlli1R4LUUh674Q79DjSOSegdIiI68U",
+  "c0-R2wvW1yRnRjQdUqetgD9tDJSCGpeJjz1HthfXwQ8",
+  "wsT2snFHYQ7AX7OxnrFViyu4v5il6sIb9EYxTnBnMQc",
+  "FyQ9uMx1XevItG1kE65BMvbbqcvdOGJrC_nb-PPIawk",
+  "xswbZRtkjQQ8D1h6tx503iLaAxxPLWP10J2TvgbRZXk",
+  "NQy9H6oAE-m55BheXbGu70nEWiiGMsL8lM9YsNJ8gD4",
+  "gkcnuAZeFeqPvFvNABFKGRKGE_AsmA0T3I1_jOFF0MU",
+]);
+
+const manualClaimableFlpIds = new Set([
+  "NXZjrPKh-fQx8BUCG_OXBUtB4Ix8Xf0gbUtREFoWQ2Q",
+  "rW7h9J9jE2Xp36y4SKn2HgZaOuzRmbMfBRPwrFFifHE",
+  "3eZ6_ry6FD9CB58ImCQs6Qx_rJdDUGhz-D2W1AqzHD8",
+  "Wc8Rg-owsWSvrmb5XAlmSs3_4UtHo9i5ui2o9UCFuTk",
+]);
```

Comment:
Can we create constants and name them or at least add comments so they are easier to update if necessary?